### PR TITLE
feature/ladder

### PR DESCRIPTION
**Pull request by NathanCenter**
## Changes:
Added the Rivet player ability to climb a ladder

## How to test:
go to a ladder that is near the player as Rivet( please quit the game and pick rivet as the person you want to play)
press the w key to move up the ladder and the s key to go down.
Move up and down and then jump off the ladder.
See if the animations look ok or need improvement.
## Sources cited:
https://www.youtube.com/watch?v=njY3NFnAI8o

### CODE REVIEW

File: src/scripts/world/ladder.gd [hunk 1]
```diff
@@ -0,0 +1,15 @@
+extends Node3D
+
+
+
+
+func _on_static_body_3d_body_entered(body):
+	if body.name == "player":
+			print("enter ladder")
+			body.at_ladder = true
+
```
Comment (JoonasHeinonen) [inline src/scripts/world/ladder.gd:10]: Remove this emptyline

File: src/scripts/world/ladder.gd [hunk 2]
```diff
@@ -0,0 +1,15 @@
+extends Node3D
+
+
+
+
```
Comment (JoonasHeinonen) [inline src/scripts/world/ladder.gd:5]: Remove these empty lines.

File: src/scripts/world/ladder.gd [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+func _on_static_body_3d_body_entered(body):
+	if body.name == "player":
+			print("enter ladder")
```
Comment (JoonasHeinonen) [inline src/scripts/world/ladder.gd:8]: Remove print.

File: src/scripts/world/ladder.gd [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+			body.at_ladder = true
+
+
+func _on_static_body_3d_body_exited(body):
+		if body.name  == "player":
+			print("exit ladder")
```
Comment (JoonasHeinonen) [inline src/scripts/world/ladder.gd:14]: Remove print.

File: src/scripts/Player/player.gd [hunk 5]
```diff
@@ -281,7 +282,38 @@ func _physics_process(delta):
...
+	
+	if Input.is_action_pressed("ui_climb_down") and at_ladder == true:
+		gravity = 0
+		player_velocity.x = 0
+		player_velocity.y = -3
+		if Globle.player_character == "Rivet":
+			$RivetSprite.hide()
+			$RivetClimingSprite.show() 
+			$RivetAnimationPlayer.play("Player_Climb_Down")
+	if Input.is_action_just_released("ui_climb_up") and at_ladder == true:
```
Comment (JoonasHeinonen) [inline src/scripts/Player/player.gd:304]: Add a space above this line.

File: src/scripts/Player/player.gd [hunk 6]
```diff
@@ -281,7 +282,38 @@ func _physics_process(delta):
...
+		player_velocity.y = 0
+		if Globle.player_character == "Rivet":
```
Comment (JoonasHeinonen) [inline src/scripts/Player/player.gd:306]: Add a space above this line.

File: src/scripts/Player/player.gd [hunk 7]
```diff
@@ -281,7 +282,38 @@ func _physics_process(delta):
...
+			$RivetSprite.hide()
+			$RivetClimingSprite.show() 
+			$RivetAnimationPlayer.play("Player_Climb_Idle")
+	if Input.is_action_just_released("ui_climb_down") and at_ladder == true:
```
Comment (JoonasHeinonen) [inline src/scripts/Player/player.gd:310]: Add a space above this line.

File: src/scripts/Player/player.gd [hunk 8]
```diff
@@ -281,7 +282,38 @@ func _physics_process(delta):
...
+		player_velocity.y = 0
+	if at_ladder == false:
```
Comment (JoonasHeinonen) [inline src/scripts/Player/player.gd:312]: Add a space above this line.

File: src/scripts/Player/player.gd [hunk 9]
```diff
@@ -281,7 +282,38 @@ func _physics_process(delta):
...
+		gravity = 3
+		if Globle.player_character == "Rivet":
+			$RivetClimingSprite.hide()
+			$RivetSprite.show()
 	if Input.is_action_pressed("ui_ranged_sniper_aim") && !Input.is_action_pressed("ui_melee_attack"):
```
Comment (JoonasHeinonen) [inline src/scripts/Player/player.gd:317]: Add a space above this line.

File: src/scripts/Player/player.gd [hunk 10]
```diff
@@ -281,7 +282,38 @@ func _physics_process(delta):
 
 	if Input.is_action_just_released("ui_accept"):
 		Globle.update_vendor()
-
+	# Ladder logic
+	if Input.is_action_pressed("ui_climb_up") and at_ladder == true:
+		gravity = 0
+		player_velocity.x = 0
+		player_velocity.y = 3
+		print(player_velocity.y)
```
Comment (JoonasHeinonen) [inline src/scripts/Player/player.gd:290]: Remove print.

File: src/scripts/Player/player.gd [hunk 11]
```diff
@@ -164,7 +165,7 @@ func _physics_process(delta):
 			for audio_sub_child in audio_container_child.get_children():
 				for audio_child in audio_sub_child.get_children():
 					audio_child.position = Vector3(self.position.x, self.position.y, 0)
-
+
```
Comment (JoonasHeinonen) [inline src/scripts/Player/player.gd:168]: Remove these empty spaces, keep the line empty.

**Review by JoonasHeinonen — CHANGES_REQUESTED** [review]
Minor changes required.

**Review by JoonasHeinonen — APPROVED** [review]
LGTM.

**Review by JoonasHeinonen — CHANGES_REQUESTED** [review]
I will finish this...

**Comment by NathanCenter** [comment]
> I will finish this...

um... what seems to be the issue is it the animation or the code?

**Comment by NathanCenter** [comment]
Looks good to me.

**Review by JoonasHeinonen — APPROVED** [review]
LGTM.

**Pull request merged**